### DISCUSSION TITLE
How to cache the result of resolver chains

### DISCUSSION THREAD

**Discussion by itssamuelrowe** [apollographql/apollo-server#6378] [Q&A]
Hi.

Let's say I have a schema like this:

```
type Account {
   ...
}

type Transaction {
    account: Account!
    ...
}
```

The equivalent resolvers would be:

```
Query: {
 ...
},
Transaction: {
   account: async (...) => {
     // Make a database call
   }
}
```

I am trying to fetch 100 `Transaction` objects all pointing to the same `Account` object. Apollo invokes the chain resolver `Transaction.account` 100 times despite the account being the same.

Do I have to manually implement a caching strategy? Or is Apollo Server capable of doing this? If yes, how do I implement it?

Thanks!

**Comment by glasser** [comment]
This is what DataLoader is designed for: https://www.npmjs.com/package/dataloader
Typically you set up a DataLoader in your `context` function and call it from your resolvers.
